core/trino-main/src/main/java/io/trino/execution/scheduler/FaultTolerantStageScheduler.java

```diff
@@ -637,7 +638,12 @@ private void updateTaskStatus(TaskStatus taskStatus, Optional<ExchangeSinkInstan
                                 log.debug("Computed next memory requirements for task from stage %s; previous=%s; new=%s; peak=%s; estimator=%s", stage.getStageId(), memoryLimits, newMemoryLimits, taskStatus.getPeakMemoryReservation(), partitionMemoryEstimator);
 
                                 if (errorCode != null && isOutOfMemoryError(errorCode) && newMemoryLimits.getRequiredMemory().toBytes() * 0.99 <= taskStatus.getPeakMemoryReservation().toBytes()) {
-                                    failure = new TrinoException(() -> errorCode, "Cannot allocate enough memory for task", failureInfo.toException());
+                                    String message = format(
+                                            "Cannot allocate enough memory for task %s. Reported peak memory reservation: %s. Computed next memory requirement: %s.",
```

Comment:
Not sure if `Computed next memory requirement:` is not too much of an implementation detail to be part of final error message. Maybe change to `Maximum possible reservation:`

Reply:
It makes sense. Let me change it.